modules/saliency/samples/computeSaliency.cpp

```diff
@@ -150,11 +150,28 @@ int main( int argc, char** argv )
       saliencyAlgorithm = ObjectnessBING::create();
       vector<Vec4i> saliencyMap;
       saliencyAlgorithm.dynamicCast<ObjectnessBING>()->setTrainingPath( training_path );
-      saliencyAlgorithm.dynamicCast<ObjectnessBING>()->setBBResDir( training_path + "/Results" );
```

Comment:
Unfortunately on Linux output file can't be created properly.
Please change zero [here](https://github.com/opencv/opencv_contrib/blob/933f53e47fe93b8318f17aa1878469923d62b1d6/modules/saliency/src/BING/CmFile.cpp#L75) and few lines below to `S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH` to fix it.

Reply:
BTW, This fix could be postponed - some filesystem utility functions will be added into OpenCV's core module including recursive directories creation.

Reply:
sure. while it could be postponed, having the correct flags here won't do any harm in the meantime.